File: remove_duplicate.c
int removeDuplicates(int* nums, int numsSize){
    if(numsSize == 0){
        return 0;
    }
    int index = 0;
    int cur = 0;
    while(index<numsSize){
        if(nums[cur]==nums[index]){
            index++;
        }else{
            cur++;
            nums[cur]=nums[index++];
        }
    }
    return cur+1;
}

File: remove_duplicate.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        index=0
        cur = 0
        while(index<len(nums)):
            if nums[cur] == nums[index]:
                index+=1
            else:
                cur+=1;
                nums[cur]=nums[index]
                index+=1       
        return cur+1